jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/GenericJdbcDatabaseAccessor.java

```diff
@@ -108,59 +143,34 @@ private <T> List<T> getColumnMetadata(Configuration conf, ColumnMetadataAccessor
 
   }
 
+  private <T> List<T> getColumnMetadata(ResultSet rs, ColumnMetadataAccessor<T> colAccessor)
```

Comment:
Do we really need this method? we already have getColumnMetadata. I like the idea of making methods smaller and more comprehensive, in this case I feel we dont really need this method.
This also introduces a need to throw a generic exception instead a specific one which should be avoided.

Reply:
There are some places need to extract the result metadata from ResultSet, 

https://github.com/apache/hive/blob/a838ca2760799708c465051904b96cab4ae628d1/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/JdbcRecordIterator.java#L69-L70

https://github.com/apache/hive/blob/a838ca2760799708c465051904b96cab4ae628d1/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/GenericJdbcDatabaseAccessor.java#L135

Method is declared as private, the upper caller will deal with the specific exception. For example:
https://github.com/apache/hive/blob/a838ca2760799708c465051904b96cab4ae628d1/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/GenericJdbcDatabaseAccessor.java#L250-L254

https://github.com/apache/hive/blob/a838ca2760799708c465051904b96cab4ae628d1/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/JdbcRecordIterator.java#L81-L84

Reply:
Makes sense thanks for the explaination